src/unix/atomic-ops.h

```diff
@@ -49,8 +48,10 @@ UV_UNUSED(static int cmpxchgi(int* ptr, int oldval, int newval)) {
     return oldval;
   else
     return op4;
+#elif defined(__SUNPRO_C) || defined(__SUNPRO_CC)
+  return atomic_cas_uint(ptr, oldval, newval);
 #else
-  return __sync_val_compare_and_swap(ptr, oldval, newval);
+#error "Missing compare-and-swap implementation."
```

Comment:
I don't think we want to do that, `__sync_val_compare_and_swap(ptr, oldval, newval);` is our "catch all" so the GCC builtins kick in.